utils/wasm-builder/src/crate_info.rs

```diff
@@ -67,4 +68,33 @@ impl CrateInfo {
             .iter()
             .find(|package| package.id == *root_id)
     }
+
+    /// check package
+    ///
+    /// - if crate-type contains "lib" or "rlib":
+    fn check(pkg: &Package) -> StdResult<&Package, BuilderError> {
+        // cargo can't import executables (bin, cdylib etc), but libs
+        // only (rlib).
+        //
+        // if no `[lib]` table, the `crate_types` will be [ "lib" ]
+        // by default, we can not detect if this is "rlib" because it
+        // is the "compiler recommended" style of library.
+        //
+        // see also https://doc.rust-lang.org/reference/linkage.html
+        let lib_s = "lib".to_string();
+        let rlib_s = "rlib".to_string();
```

Comment:
We can compare with string slices too.
```suggestion
        let lib_s = "lib";
        let rlib_s = "rlib";
```